modules/sources/contracts/v3/fetch-unwrap-rates.ts

```diff
@@ -0,0 +1,24 @@
+import { Chain } from '@prisma/client';
+import { Multicaller3Viem } from '../../../web3/multicaller-viem';
+import MinimalErc4626Abi from '../abis/MinimalERC4626';
+import { formatEther, parseEther } from 'viem';
+
+/**
+ * Fetches convertToAssets rates for a list of ERC4626 tokens and returns them as floats
+ *
+ * @param addresses
+ * @param chain
+ * @returns
+ */
+export const fetchUnwrapRates = async (addresses: string[], chain: Chain) => {
+    const caller = new Multicaller3Viem(chain, MinimalErc4626Abi);
+    addresses.forEach((address) => caller.call(address, address, 'convertToAssets', [parseEther('1')]));
```

Comment:
@johngrantuk - I'd like to discuss this line with you. 
I decided to use 1e18 as input to `convertToAssets` so we always get the equivalent of 18 decimals `unwrapRate` disregard of the original token decimals (unless I'm missing something 🤔). 
This should make it easier to handle any math regarding rates, right?
I know you took a different approach on Paraswap PR, so it would be nice to hear your opinion.

Reply:
If you do it that way you will then have to scale the rate to the raw decimals when passing it as `rate` for a `Buffer` pool. That way the maths will work.

Reply:
So, let me check if I got this right:
1. balancer maths expect rate to be passed in with decimals equals to the erc4626 (wrapped) token
2. internally balancer maths always scale that rate to 18 decimals (possibly losing precision on RAY math implementations)

Ps: thanks for the test info - I'll create some tests to make sure the SOR is working as expected

Reply:
@brunoguerios - sorry I read this wrong! Always using 1e18 for `convertToAssets` is correct. You should just be able to pass the returned rate as the buffer rate to the maths and it will work.

Reply:
This would be so much easier if everyone kept consistency between wrapped/underlying decimals. This one case of 18 decimals wrapped and 6 decimals underlying will require quite a big change 🙄